visualization/app/codeCharta/ui/artificialIntelligence/artificialIntelligence.component.ts

```diff
@@ -139,80 +142,80 @@ export class ArtificialIntelligenceController
-		let totalRloc = 0
-		let numberOfRlocLowRisk = 0
-		let numberOfRlocModerateRisk = 0
-		let numberOfRlocHighRisk = 0
-		let numberOfRlocVeryHighRisk = 0
-
-		const languageSpecificThresholds = this.getAssociatedMetricThresholds(programmingLanguage)
-		const thresholds = languageSpecificThresholds[metricName]
-
+	private calculateRiskProfile(fileState: FileState, metricName: string) {
 		for (const { data } of hierarchy(fileState.file.map)) {
 			// TODO calculate risk profile only for focused or currently visible but not excluded files.
-			if (
-				data.type !== NodeType.FILE ||
-				isPathBlacklisted(data.path, this.blacklist, BlacklistType.exclude) ||
-				data.attributes[metricName] === undefined ||
-				data.attributes["rloc"] === undefined ||
-				this.getFileExtension(data.name) !== programmingLanguage
-			) {
-				continue
-			}
-
-			const nodeMetricValue = data.attributes[metricName]
-			const nodeRlocValue = data.attributes["rloc"]
-
-			totalRloc += nodeRlocValue
-
-			// Idea: We could calculate risk profiles per directory in the future.
-			if (nodeMetricValue <= thresholds.percentile70) {
-				numberOfRlocLowRisk += nodeRlocValue
-			} else if (nodeMetricValue <= thresholds.percentile80) {
-				numberOfRlocModerateRisk += nodeRlocValue
-			} else if (nodeMetricValue <= thresholds.percentile90) {
-				numberOfRlocHighRisk += nodeRlocValue
-			} else {
-				numberOfRlocVeryHighRisk += nodeRlocValue
+			if (this.isFileInvalid(data, metricName)) {
+				const fileExtension = this.getFileExtension(data.name)
+				const languageSpecificThresholds = this.getAssociatedMetricThresholds(fileExtension)
+				const thresholds = languageSpecificThresholds[metricName]
+				const nodeMetricValue = data.attributes[metricName]
+				const nodeRlocValue = data.attributes[AREA_METRIC]
+
+				// Idea: We could calculate risk profiles per directory in the future.
+				this.calculateRlocRisk(nodeMetricValue, thresholds, nodeRlocValue)
 			}
 		}
 
-		if (totalRloc === 0) {
+		if (this.rlocRisk.totalRloc === 0) {
 			return
 		}
 
+		this.setRiskProfile()
+	}
+
+	private setRiskProfile() {
 		const [lowRisk, moderateRisk, highRisk, veryHighRisk] = percentRound([
-			numberOfRlocLowRisk,
-			numberOfRlocModerateRisk,
-			numberOfRlocHighRisk,
-			numberOfRlocVeryHighRisk
+			this.rlocRisk.lowRisk,
+			this.rlocRisk.moderateRisk,
+			this.rlocRisk.highRisk,
+			this.rlocRisk.veryHighRisk
 		])
 
-		this._viewModel.riskProfile = {
-			lowRisk,
-			moderateRisk,
-			highRisk,
-			veryHighRisk
+		this._viewModel.riskProfile = { lowRisk, moderateRisk, highRisk, veryHighRisk }
+	}
+
+	private isFileInvalid(node: CodeMapNode, metricName: string) {
+		return (
+			node.type === NodeType.FILE &&
+			!isPathBlacklisted(node.path, this.blacklist, BlacklistType.exclude) &&
+			node.attributes[metricName] !== undefined &&
+			node.attributes[AREA_METRIC] !== undefined &&
+			!EXCLUDED_FILES.has(this.getFileExtension(node.name))
+		)
+	}
```

Comment:
I can't really follow that expression. I assume, that a file is invalid, if it's a file and one of the following things are true:
- the file is excluded
- missing a specific metric
- missing the area metric
- the file extension is on our EXCLUDED_FILES blacklist

```ts
	private isFileInvalid(node: CodeMapNode, metricName: string) {
		return node.type === NodeType.FILE && (
			isPathBlacklisted(node.path, this.blacklist, BlacklistType.exclude) ||
			node.attributes[metricName] === undefined ||
			node.attributes[AREA_METRIC] === undefined ||
			EXCLUDED_FILES.has(this.getFileExtension(node.name))
		)
	}
```

Reply:
It was a renaming mistake. Actually this method should call `isFileValid`.